### ISSUE TITLE
OpenAPI derive prefix_path from Path, not just string literal

### ISSUE THREAD

**Issue by Sebekerga** [poem-web/poem#635]
## Description of the feature

Right now it is not possible to use constants as prefix_path in OpenAPI macro, since this arg type is set to String. It would be nice to be able to use constants for shared prefixes, making it more typo safe.

I think, I can make those changes, but before diving in, I want to know:
1. Would that even be desirable, since it makes it impossible to normalize meta path in macro?
2. Would maintainers help me with code review? I just never contributed to large projects, so I'm a little confused as to how that's gonna go.

## Code example 

```rust
const COMMON_PREFIX: &str = "/some_cmn";

#[OpenApi(prefix_path = COMMON_PREFIX)]
impl Api1 {
    #[oai(path = "/a", method = "get")]
    async fn test(&self) {}
}
```

**Comment by Sebekerga** [comment]
I did some experiments with `OpenApi` macro, and for "dynamic" prefix_path to work some changes would have to be made:
1) Changes in macro, so that `convert_oai_path` returns `TokenStream`;
2) Changes in poem-openapi crate, so that metadata path type is not `&'static str` but `String`;

**Comment by sunli829** [comment]
I'll handle it, this is more troublesome.

**Comment by sunli829** [comment]
released in `v3.0.4`

https://github.com/poem-web/poem/blob/582329d2b5edd4121c743fefc71947ca477d1ac0/poem-openapi/tests/api.rs#L112

**Comment by Sebekerga** [comment]
Thanks a lot!
